src/Toolbar/Toolbar.react.js

```diff
@@ -112,13 +108,12 @@ function getStyles(props, context) {
 }
 
 class Toolbar extends PureComponent {
-  constructor(props) {
-    super(props);
+  constructor() {
+    super(...arguments);
+    autobind(this);
 
-    const isSearchActive = props.isSearchActive || false;
-    this.backButtonListener = isSearchActive
-      ? getBackButtonListener(this.onSearchCloseRequested)
-      : null;
+    const isSearchActive = this.props.isSearchActive || false;
+    BackAndroid.addEventListener('hardwareBackPress', this._onBackPress);
```

Comment:
should we add this only if the search is active? because with this, the listener will always be added whenever the Toolbar is rendered right?